scheduler/scheduler-server/src/test/java/functionaltests/utils/SchedulerTHelper.java

```diff
@@ -759,41 +759,15 @@ public JobInfo waitForEventJobFinished(JobId id) throws Exception {
      * @throws Exception
      */
     public JobInfo waitForEventJobFinished(Scheduler userInterface, JobId id) throws Exception {
-        try {
-            JobEventMonitor monitorRunningToFinished = null;
-            JobEventMonitor monitorPendingToFinished = null;
-            synchronized (getSchedulerMonitorsHandler()) {
-                Optional<JobEventMonitor> opJobMonitor = getSchedulerMonitorsHandler().removeJobEvent(id,
-                                                                                                      SchedulerEvent.JOB_RUNNING_TO_FINISHED);
-                if (opJobMonitor.isPresent()) {
-                    //event occurred, remove it and return associated Job object
-                    return opJobMonitor.get().getJobInfo();
-                }
-                opJobMonitor = getSchedulerMonitorsHandler().removeJobEvent(id, SchedulerEvent.JOB_PENDING_TO_FINISHED);
-                if (opJobMonitor.isPresent()) {
-                    //event occurred, remove it and return associated Job object
-                    return opJobMonitor.get().getJobInfo();
-                }
-
-                monitorRunningToFinished = (JobEventMonitor) getSchedulerMonitorsHandler().getMonitor(new JobEventMonitor(SchedulerEvent.JOB_RUNNING_TO_FINISHED,
-                                                                                                                          id));
-                monitorPendingToFinished = (JobEventMonitor) getSchedulerMonitorsHandler().getMonitor(new JobEventMonitor(SchedulerEvent.JOB_PENDING_TO_FINISHED,
-                                                                                                                          id));
-            }
-
-            getSchedulerMonitorsHandler().waitWithAnyMonitor(0, monitorRunningToFinished, monitorPendingToFinished);
-
-            if (monitorRunningToFinished.eventOccured()) {
-                return monitorRunningToFinished.getJobInfo();
-            } else if (monitorPendingToFinished.eventOccured()) {
-                return monitorPendingToFinished.getJobInfo();
-            } else {
-                // erorr
-                return null;
+        JobState jobState = null;
+        while (jobState == null || !jobState.isFinished()) {
+            try {
+                jobState = getSchedulerInterface().getJobState(id);
+            } catch (UnknownJobException e) {
             }
-        } catch (ProActiveTimeoutException e) {
-            return null;
+            Thread.sleep(1000);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this use of "Thread.sleep()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)